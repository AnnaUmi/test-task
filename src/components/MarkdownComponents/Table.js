import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Table = styled.table`
  font-style: normal;
  font-weight: 400;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  color: rgb(51, 51, 51);
  font-family: Roboto, sans-serif;
  font-size: 16px;
  border-collapse: collapse;
  width: 100%;
  border-bottom: none;
`;

const TH = styled.th`
  font-style: normal;
  font-weight: bold;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  color: rgb(51, 51, 51);
  font-family: Roboto, sans-serif;
  font-size: 16px;
  line-height: 1.44;
  text-align: left;
  vertical-align: top;
  padding: 16px 16px 16px 0;
`;

const TD = styled.th`
  font-style: normal;
  font-weight: 400;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  color: rgb(51, 51, 51);
  font-family: Roboto, sans-serif;
  font-size: 16px;
  line-height: 1.44;
  text-align: left;
  vertical-align: top;
  padding: 16px 16px 16px 0;
`;

const P = styled.p`
  font-style: normal;
  font-weight: 400;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  color: rgb(51, 51, 51);
  font-family: Roboto, sans-serif;
  font-size: 16px;
  line-height: 1.44;
  flex-basis: 100%;
  margin: 0;
`;

const THead = styled.thead`
  border-bottom: 2px solid rgb(214, 214, 214);
`;

const TR = styled.tr`
  border-bottom: 1px solid rgb(214, 214, 214);
`;

const TableComp = props => <Table {...props} />;

const THElement = props => {
  const { children, colspan, rowspan } = props;
  return (
    <TH colSpan={colspan} rowSpan={rowspan}>
      {children && children.props && children.props.children}
    </TH>
  );
};

THElement.propTypes = {
  children: PropTypes.node.isRequired,
  colspan: PropTypes.number,
  rowspan: PropTypes.number,
};

THElement.defaultProps = {
  colspan: 1,
  rowspan: 1,
};

const TDElement = props => {
  const { children, colspan, rowspan } = props;
  return (
    <TD colSpan={colspan} rowSpan={rowspan}>
      <P>{children && children.props && children.props.children}</P>
    </TD>
  );
};

TDElement.propTypes = {
  children: PropTypes.node,
  colspan: PropTypes.number,
  rowspan: PropTypes.number,
};

TDElement.defaultProps = {
  children: null,
  colspan: 1,
  rowspan: 1,
};

const THeadElement = props => <THead {...props} />;
const TRElement = props => <TR {...props} />;

export { TableComp, THElement, THeadElement, TRElement, TDElement };
