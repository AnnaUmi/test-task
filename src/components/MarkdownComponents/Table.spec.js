import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { TableComp, THElement, THeadElement, TRElement, TDElement } from './Table';

describe('Table', () => {
  it('TableComp should match snapshot', () => {
    const tree = renderer.create(<TableComp />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('THElement should match snapshot', () => {
    const props = {
      colspan: 3,
      rowspan: 3,
      children: {
        props: {
          children: 'test data',
        },
      },
    };
    const tree = renderer.create(<THElement {...props} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('THeadElement should match snapshot', () => {
    const tree = renderer.create(<THeadElement />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('TRElement should match snapshot', () => {
    const tree = renderer.create(<TRElement />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('TDElement should match snapshot when we pass children', () => {
    const props = {
      colspan: 3,
      rowspan: 3,
      children: {
        props: {
          children: 'test data',
        },
      },
    };
    const tree = renderer.create(<TDElement {...props} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
