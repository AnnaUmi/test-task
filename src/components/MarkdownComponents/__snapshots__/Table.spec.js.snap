// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table TDElement should match snapshot when we pass children 1`] = `
.c0 {
  font-style: normal;
  font-weight: 400;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  color: rgb(51,51,51);
  font-family: Roboto,sans-serif;
  font-size: 16px;
  line-height: 1.44;
  text-align: left;
  vertical-align: top;
  padding: 16px 16px 16px 0;
}

.c1 {
  font-style: normal;
  font-weight: 400;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  color: rgb(51,51,51);
  font-family: Roboto,sans-serif;
  font-size: 16px;
  line-height: 1.44;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  margin: 0;
}

<th
  className="c0"
  colSpan={3}
  rowSpan={3}
>
  <p
    className="c1"
  >
    test data
  </p>
</th>
`;

exports[`Table THElement should match snapshot 1`] = `
.c0 {
  font-style: normal;
  font-weight: bold;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  color: rgb(51,51,51);
  font-family: Roboto,sans-serif;
  font-size: 16px;
  line-height: 1.44;
  text-align: left;
  vertical-align: top;
  padding: 16px 16px 16px 0;
}

<th
  className="c0"
  colSpan={3}
  rowSpan={3}
>
  test data
</th>
`;

exports[`Table THeadElement should match snapshot 1`] = `
.c0 {
  border-bottom: 2px solid rgb(214,214,214);
}

<thead
  className="c0"
/>
`;

exports[`Table TRElement should match snapshot 1`] = `
.c0 {
  border-bottom: 1px solid rgb(214,214,214);
}

<tr
  className="c0"
/>
`;

exports[`Table TableComp should match snapshot 1`] = `
.c0 {
  font-style: normal;
  font-weight: 400;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  color: rgb(51,51,51);
  font-family: Roboto,sans-serif;
  font-size: 16px;
  border-collapse: collapse;
  width: 100%;
  border-bottom: none;
}

<table
  className="c0"
/>
`;
